mama_ng_control/apps/subscriptions/tasks.py

```diff
@@ -1,9 +1,76 @@
-from celery import task
+from celery.task import Task
 from celery.utils.log import get_task_logger
+from celery.exceptions import SoftTimeLimitExceeded
+from client.messaging_contentstore.contentstore import ContentStoreApiClient
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
 
 logger = get_task_logger(__name__)
 
+from mama_ng_control.apps.vumimessages.models import Outbound
+from mama_ng_control.apps.contacts.models import Contact
+
+
+class Create_Message(Task):
+
+    """
+    Task to create and populate a message with content
+    """
+    name = "mama_ng_control.apps.subscriptions.tasks.create_message"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def contentstore_client(self):
+        return ContentStoreApiClient(
+            auth_token=settings.CONTENTSTORE_AUTH_TOKEN,
+            api_url=settings.CONTENTSTORE_API_URL)
+
+    def run(self, contact_id, messageset_id, sequence_number, lang,
+            subscription, **kwargs):
+        """
+        Returns success message
+        """
+        l = self.get_logger(**kwargs)
+        l.info("Creating Outbound Message and Content")
+        try:
+            contact = Contact.objects.get(pk=contact_id)
+            contentstore = self.contentstore_client()
+            params = {
+                "messageset": messageset_id,
+                "sequence_number": sequence_number,
+                "lang": lang
+            }
+            # should only return one in a list
+            messages = contentstore.get_messages(params=params)
+            if len(messages) > 0:
+                message_id = messages[0]["id"]
```

Comment:
I'd add a comment here to explain that if contentstore is badly managed and there is more than one message it'll just grab the first one.